benchmarks/canny_edge_detector/procedural_f#/Program.fs

```diff
@@ -0,0 +1,222 @@
+﻿// Learn more about F# at http://fsharp.org
+
+open System
+open System.Drawing
+
+let weak = 100
+let black = 0
+let white = 255
+let getPixel (image: Bitmap) (x: int) (y: int) = (int)(image.GetPixel(x, y).R)
+
+let kernelHor = array2D [
+    [-1.0; 0.0; 1.0]
+    [-2.0; 0.0; 2.0]
+    [-1.0; 0.0; 1.0]]
+
+let kernelVer = array2D [
+    [1.0; 2.0; 1.0]
+    [0.0; 0.0; 0.0]
+    [-1.0; -2.0; -1.0]]
+
+
+// https://epochabuse.com/gaussian-blur/
+let GaussianFilter (length: int) (weight: float) = 
+    let kernel = Array2D.create length length 0.0
+    let mutable kernelSum = 0.0
+    let foff = (length - 1) / 2
+    let mutable distance = 0.0
+    let constant = 1.0 / (2.0 * Math.PI * weight * weight)
+    for y in -foff .. foff do
+        for x in -foff .. foff do
+            distance <- (float)((y * y) + (x * x)) / (2.0 * (weight * weight))
+            kernel.[y + foff, x + foff] <- constant * Math.Exp(-distance)
+            kernelSum <- kernelSum + kernel.[y + foff, x + foff]
+
+    for y in 0 .. length-1 do
+        for x in 0 .. length-1 do
+            kernel.[y, x] <- kernel.[y, x] * 1.0 / kernelSum
+    kernel
+
+//My own slow Convolve
+let Convolve (image: int[,]) (filter: float[,]) =
+    let width = image.GetLength(0)
+    let height = image.GetLength(1)
+    //Kernel has to be an odd number
+    let halfKernel = filter.GetLength(0) / 2
+    
+    let test: int[,] = Array2D.zeroCreate width height
+    for x in 0 .. width-1 do
+        for y in 0 .. height-1 do
```

Comment:
Hvis du kører til width-halfkernel og height-halfkernel slipper du for if'en længere nede